charts/langsmith/docs/DELETE-TRACES.md

```diff
@@ -0,0 +1,63 @@
+# Deleting Traces
+
+The LangSmith UI does not currently support the deletion of an invidual trace.  This, however, can be accomplished by directly removing the trace from all materialized views in ClickHouse (except the runs_history views) and the runs table itself.
+
+This command can either be run using a trace ID as an argument or using a file that is a list of trace IDs.
+
+### Prerequisites
+
+Ensure you have the following tools/items ready.
+
+1. Kubectl 
+
+## Migrating old runs locally
+
+### Prerequisites
+
+Ensure you have the following tools/items ready.
+
+1. Clickhouse database credentials
+    - Host
+    - Port
+    - Username
+      - If using the bundled version, this is `default`
+    - Password
+      - If using the bundled version, this is `password`
+    - Database name
+       - If using the bundled version, this is `default`
+    
+2. Connectivity to the Clickhouse database from the machine you will be running the `delete_trace_by_id` script on.
+   - If you are using the bundled version, you may need to port forward the clickhouse service to your local machine.
+   - Run `kubectl port-forward svc/langsmith-clickhouse 8123:8123` to port forward the clickhouse service to your local machine.
+
+### Running the deletion script for a single trace
+
+Run the following command to run the migration script:
+
+```bash
+sh delete_trace_by_id.sh <clickhouse_url> --trace_id <trace_id> 
+```
+
+For example, if you are using the bundled version with port-forwarding, the command would look like:
+
+```bash
+sh delete_trace_by_id.sh "clickhouse://default:password@localhost:8123/default" --trace_id 4ec70ec7-0808-416a-b836-7100aeec934b
+```
+
+If you visit the Langsmith UI, you should now see trace ID is deleted.
+
+### Running the deletion script for a multiple traces from a file with one trace ID per line
+
+Run the following command to run the migration script:
+
+```bash
+sh delete_trace_by_id.sh <clickhouse_url> --file <path/to/foo.txt> 
+```
+
+For example, if you are using the bundled version with port-forwarding, the command would look like:
+
+```bash
+sh delete_trace_by_id.sh "clickhouse://default:password@localhost:8123/default" --file path/to/traces.txt
```

Comment:
is there a reason to use https vs native port and ch cli here? just less dependencies?

Reply:
Yup, dependency reduction.  Clickhouse package is a little bit odd to install and since there is always an http interface...